### PR TITLE
feat(providers): Add Loops Email Provider and Documentation

### PR DESCRIPTION
**Pull request by Whats-A-MattR**
## ☕️ Reasoning

Email / Magic Token Auth is great.
Loops.so make a great product for Email that goes beyond just transactional emails, I use them as my provider and love the platform. 

## 🧢 Checklist

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged

## 🎫 Affected issues

Fixes: [This Feature Request](https://github.com/nextauthjs/next-auth/discussions/11196)

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Comment by Whats-A-MattR** [comment]
Pls approve.

**Comment by ndom91** [comment]
Thanks, this looks great! I made some tiny edits regarding the `loops.svg` image file. 

Could you update the svelte examples regarding the environment variables? Unfortunately SvelteKit doesn't expose a `process.env`. You have to either import `env` or the variables themselves. [More details](). I realize this needs updated on the other magic link providers docs as well.

Soemthing like:

```ts
import { SvelteKitAuth } from "@auth/sveltekit"
import Loops from "@auth/sveltekit/providers/loops"
import { env } from "$env/dynamic/prviate"

export const { handle, signIn, signOut } = SvelteKitAuth({
  providers: [
    Loops({
      apiKey: env.AUTH_LOOPS_KEY,
      transactionalId: env.AUTH_LOOPS_TRANSACTIONAL_ID,
    })
  ]
})
```

**Comment by Whats-A-MattR** [comment]
Hey, @ndom91 I've updated in-line with the docs provided by Svelte 
https://learn.svelte.dev/tutorial/env-static-private

**Comment by Whats-A-MattR** [comment]
@ndom91 Hey, how are we feeling about merging this one?

File: packages/core/src/providers/loops.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    theme: Theme
+    request: Request
+  }) => Awaitable<void>
+  options: LoopsUserConfig
+}
+
+export type LoopsUserConfig = Omit<Partial<LoopsConfig>, "options" | "type">
+
+export default function Loops(config: LoopsUserConfig): LoopsConfig {
+  if (!config.transactionalId)
```
Comment (ndom91) [inline packages/core/src/providers/loops.ts:33]: If `transactionalId` is required, let's not make it optional in the types 👍

Comment (Whats-A-MattR) [reply packages/core/src/providers/loops.ts:33]: updated 😄 
thanks for that, good catch

**Review by Whats-A-MattR — COMMENTED** [review]
updated type for mandatory transactionId

**Review by ndom91 — APPROVED** [review]
Great, LGTM! Thanks :pray:

**Comment by ndom91** [comment]
Looks like there are still some TS issues [here](https://github.com/nextauthjs/next-auth/actions/runs/10603214620/job/29387015818?pr=11197) though

**Review by Whats-A-MattR — COMMENTED** [review]
ready for merge

File: packages/core/src/providers/email.ts [hunk 2]
```diff
@@ -26,21 +26,24 @@ export default function Email(config: NodemailerUserConfig): NodemailerConfig {
...
-  sendVerificationRequest: (params: {
-    identifier: string
-    url: string
-    expires: Date
-    provider: EmailConfig
-    token: string
-    theme: Theme
-    request: Request
-  }) => Awaitable<void>
+  /** updated by Whats-A-MattR to allow extension for mail vendors that require more than only an API key */
```
Comment (ndom91) [inline packages/core/src/providers/email.ts:45]:
```suggestion
```

**Comment by ndom91** [comment]
@Whats-A-MattR just wanted to simplify this a bit further as all those extra interfaces weren't necessary. Thanks for getting it back to building :pray:

**Comment by ndom91** [comment]
Ah so after fixing some of the other build issues in the repo, it looks like there's still an issue with this PR / loops provider, see: https://github.com/nextauthjs/next-auth/actions/runs/10983054627/job/30492001767?pr=11197

**Comment by Whats-A-MattR** [comment]
> Ah so after fixing some of the other build issues in the repo, it looks like there's still an issue with this PR / loops provider, see: https://github.com/nextauthjs/next-auth/actions/runs/10983054627/job/30492001767?pr=11197

yeah, I had implemented different types for Loops but you removed them in [this commit](https://github.com/nextauthjs/next-auth/pull/11197/commits/f8d9fd9ee9ada4cbcd1b8a0562c6d2f58c897ba1)

I've now resolved and build is working again - at least locally...

**Comment by Whats-A-MattR** [comment]
How about we get some merge happening @ndom91 @ThangHuuVu @ubbe-xyz 
Should all be ready to go

**Comment by Whats-A-MattR** [comment]
> How about we get some merge happening @ndom91 @ThangHuuVu @ubbe-xyz Should all be ready to go

bumping again

**Comment by Whats-A-MattR** [comment]
updating from base (again).

**Comment by Whats-A-MattR** [comment]
@ndom91 @ThangHuuVu @ubbe-xyz could we possibly do the thing now please?

**Comment by Whats-A-MattR** [comment]
> >  @ndom91 @ThangHuuVu @ubbe-xyz could we possibly do the thing now please?
> 
> Please don't tag maintainers directly 🙏 
> 
> Regarding getting this in, are the extra types you added back really necessary? Look at how simple the other email providers are for example. 


To include the transactionalId without the LSP complaining, type errors being thrown, and the build and test scripts failing I had to do the typing. 

The other providers are unaffected as all I did was type the props of a function in the interface.

**Review by ndom91 — APPROVED** [review]
Okay sounds good, lets go with this for now then. Sorry for the long PR process. 

By "look at the other email providers.." all I meant was look at how simple their types are.

**Pull request merged**